### PR TITLE
Appcues Web SDK

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/176)

### CODE REVIEW

File: integrations/Appcues/browser.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  isLoaded() {
+    logger.debug("in appcues isLoaded");
+    return !!window.Appcues;
+  }
+
+  isReady() {
+    logger.debug("in appcues isReady");
+    // if (this.sendToAllDestinations) {
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:27]: is this needed?

File: integrations/Appcues/browser.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  identify(rudderElement) {
+    const {
+      traits
+    } = rudderElement.message.context;
+    const {
+      userId
+    } = rudderElement.message;
+    if (userId) {
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:47]: should we set a message for else?

File: integrations/Appcues/browser.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      window.Appcues.identify(userId, traits);
+    }
+  }
+
+  track(rudderElement) {
+    const eventName = rudderElement.message.event;
+    let {
+      properties
+    } = rudderElement.message;
+    if (eventName) {
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:57]: same as user id

File: integrations/Appcues/browser.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      window.Appcues.track(eventName, properties);
+    }
+  }
+
+  page(rudderElement) {
+    const {
+      properties,
+      name,
+      category
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:66]: where is category needed?

File: integrations/Appcues/browser.js (hunk 1 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:27]: We are still discussing regarding it. So commented it for now ,might use it in future if needed

File: integrations/Appcues/browser.js (hunk 4 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:66]: Its not needed, removing it

File: integrations/Appcues/browser.js (hunk 1 above)
Comment (ruchiramoitra) [reply integrations/Appcues/browser.js:27]: ok sure... comment something like // TODO too

File: integrations/Appcues/browser.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    } = rudderElement.message;
+    window.Appcues.page(name, properties);
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:68]: what happends if properties is not present?? have you tested?

File: integrations/Appcues/browser.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+
+  track(rudderElement) {
+    const eventName = rudderElement.message.event;
+    let {
+      properties
+    } = rudderElement.message;
+    if (eventName) {
+      window.Appcues.track(eventName, properties);
```
Comment (ruchiramoitra) [inline integrations/Appcues/browser.js:58]: what happens if properties --> null?

File: integrations/Appcues/browser.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  page(rudderElement) {
+    const {
+      properties,
+      name,
+      category
+    } = rudderElement.message;
+    window.Appcues.page(name, properties);
+  }
+
+  reset() {
```
Comment (prabrishac) [inline integrations/Appcues/browser.js:71]: We don't have support of `reset` call to integrations in our sdk. Further enhancement in sdk is needed.

File: integrations/Appcues/browser.js (hunk 1 above)
Comment (prabrishac) [inline integrations/Appcues/browser.js:27]: Also check if `window.Appcues` is not undefined or null

**Review by prabrishac — CHANGES_REQUESTED** [review]
Please provide mapping in `client_server_name.js` and `integration_cname.js`

File: integrations/Appcues/browser.js (hunk 6 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:58]: It would work fine even if the properties are null

File: integrations/Appcues/browser.js (hunk 5 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:68]: The page call would work fine, yeah tested it

File: integrations/Appcues/browser.js (hunk 7 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:71]: Okay, will remove that

Comment (prabrishac) [reply integrations/Appcues/browser.js:71]: Dont remove. Keep it there.

File: integrations/Appcues/browser.js (hunk 2 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:47]: Yes, adding them

File: integrations/Appcues/browser.js (hunk 3 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:57]: Added now !

File: integrations/Appcues/browser.js (hunk 7 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:71]: Done

File: integrations/Appcues/browser.js (hunk 1 above)
Comment (desusai7) [reply integrations/Appcues/browser.js:27]: Added now

**Review by ruchiramoitra — COMMENTED** [review]
LGTM

**Review by prabrishac — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**